scheds/rust/scx_lavd/src/bpf/main.bpf.c

```diff
@@ -1247,9 +1247,16 @@ static u64 calc_lat_cri(struct task_struct *p, struct task_ctx *taskc,
 	/*
 	 * Make sure the lat_cri is non-zero.
 	 */
-	lat_cri = max(lat_cri, 1);
-	taskc->lat_cri = lat_cri;
-	return lat_cri;
+	taskc->lat_cri_self = max(lat_cri, 1);
+
+	/*
+	 * Determine latency criticality of a task in a context-aware manner by
+	 * considering which task wakes up this task. If its waker is more
+	 * latency-critcial, inherit waker's latency criticality.
+	 */
+	taskc->lat_cri = max(taskc->lat_cri_self, taskc->lat_cri_waker);
```

Comment:
I didn't measure a concrete number how often such propagation happens. However, I found that this is useful especially when task pools -- such as kworkers and game engine thread pool. Because a task in a pool can take any role dynamically, giving a little bit of more hint was helpful.

Reply:
Yeah, that makes sense.